### PR TITLE
Support MIGraphXEP to work with ROCMEP for inference on AMD GPU

### PR DESCRIPTION
**Pull request by weixingzhang**

### CODE REVIEW

**Review by ytaous — APPROVED** [review]

**Review by zhangyaobit — COMMENTED** [review]
Great!

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc [hunk 1]
```diff
@@ -128,7 +128,7 @@ void MIGraphXExecutionProvider::RegisterAllocator(std::shared_ptr<AllocatorManag
   allocator_ = allocator_manager->GetAllocator(device_id_, OrtMemTypeDefault);
   if (nullptr == allocator_) {
     AllocatorCreationInfo default_memory_info(
-        [](OrtDevice::DeviceId device_id) { return CreateHIPAllocator(device_id, onnxruntime::MIGRAPHX); }, device_id_);
+        [](OrtDevice::DeviceId device_id) { return CreateROCMAllocator(device_id, onnxruntime::CUDA); }, device_id_);
```
[the remark this answers is not in the document]
Comment (weixingzhang) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:131]: No. CUDA/CUDA_PINNED will be renamed to GPU/GPU_PINNED. Here is the PR: https://github.com/microsoft/onnxruntime/pull/5293

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc [hunk 2]
```diff
@@ -141,7 +141,7 @@ void MIGraphXExecutionProvider::RegisterAllocator(std::shared_ptr<AllocatorManag
   if (nullptr == hip_pinned_alloc) {
     AllocatorCreationInfo pinned_allocator_info(
         [](OrtDevice::DeviceId device_id) {
-          return CreateHIPPinnedAllocator(device_id, onnxruntime::MIGRAPHX_PINNED);
+          return CreateROCMPinnedAllocator(device_id, onnxruntime::CUDA_PINNED);
```
[the remark this answers is not in the document]
Comment (weixingzhang) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:144]: same as above.

**Pull request merged**